spring-kafka/src/main/java/org/springframework/kafka/support/ProducerListener.java

```diff
@@ -36,7 +36,11 @@
  * @author Endika Gutiérrez
  *
  * @see org.apache.kafka.clients.producer.Callback
+ *
+ * @deprecated since 3.0 in favor of {@link org.apache.kafka.clients.producer.ProducerInterceptor}
  */
+
+@Deprecated(since = "3.0", forRemoval = true)
```

Comment:
If we deprecate it, we have to replace its usage with the mentioned favor.
Any thoughts what could be done in the that `KafkaTemplate` for example?

I wonder if you ran your fix locally with the `./gradlew clean check`. It must show you that you still use some deprecate API.